js/views/cfi_navigation_logic.js

```diff
@@ -1278,16 +1278,19 @@ var CfiNavigationLogic = function(options) {
         _.each($elements, function ($node) {
             var isTextNode = ($node[0].nodeType === Node.TEXT_NODE);
             var $element = isTextNode ? $node.parent() : $node;
-            var visibilityPercentage = checkVisibilityByRectangles(
-                $node, true, visibleContentOffsets, frameDimensions);
-
-            if (visibilityPercentage) {
-                visibleElements.push({
-                    element: $element[0], // DOM Element is pushed
-                    textNode: isTextNode ? $node[0] : null,
-                    percentVisible: visibilityPercentage
-                });
-            }
+	    if (($element[0].tagName !== "A") || $node[0].nodeValue) {
```

Comment:
Probably yes.

1. My patch is conservative in that it ignores empty text nodes only when they are in <a> tags.
It may be that we can safely treat all empty text nodes and their enclosing elements as invisible regardless of what checkVisibilityByRectangles() tells us.

2. If we should not ignore all elements enclosing empty text nodes:
Is there a function (either a DOM element function or a Readium-specific) for identifying elements with non-empty content model?

If yes, its inverse should replace the ($element[0].tagName !== "A") test.

3. It is possible that checkVisibilityByRectangles() has a bug (a corner case) in handling empty text nodes. At the moment I do not have time to look into this.